scripts/deploy/deploy.js

```diff
@@ -586,13 +598,13 @@ async function main() {
       continue;
     }
 
-    if (!config.CONTRACTS_ADDRESSES[alias] || isProxy) {
-      config.CONTRACTS_ADDRESSES[alias] = address;
+    if (!addressesMap[alias] || isProxy) {
+      addressesMap[alias] = address;
     }
   }
 
-  console.log(`Updating config file ${configFile}`);
-  fs.writeFileSync(configFile, JSON.stringify(config, null, 2), 'utf8');
+  console.log(`Updating addresses.json ${addressesFile}`);
+  fs.writeFileSync(addressesFile, JSON.stringify(addressesMap, null, 2), 'utf8');
```

Comment:
After writing the `README.md` this seems very strange to me. The `build.js` script now generates ABIs based on Hardhat, but `addresses.json` is simply copied. Then `deploy.js` will actually output new files, which should actually be used by `build.js` to create a package output.

The output for abis and config (now addresses file location) were used by the `frontend-react` repo, but this now works differently.

@shark0der Let's chat about what will be the correct way forward, because now I think is needlessly complex. In my opinion running build and then deploy would make most sense. Now you need to run build -> deploy -> build.